### PR TITLE
[SERVICES-23] Mobile applications service

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
Splitting mobile configurations and mobile application to two independent services.
* Mobile config service should be dependent only from ceph.
* Mobile apps service should provide apps list from apps deployer.

Take a look: @Wikia/services-team 

@pchojnacki I've also merged your changes with filtering applications per hub [SERVICES-266], so please double-check if I didn't missed anything.

### CODE REVIEW

File: service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/core/Problem.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.wikia.services.mobileapplications.core;
+
+public class Problem {
```
Comment (nmonterroso) [inline service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/core/Problem.java]: I think we can remove this class altogether and use `com.wikia.pandora.http.Problem` provided in `:lib:service-core`

File: gradle/dropwizard_service.gradle [hunk 2]
```diff
@@ -3,22 +3,26 @@ apply from: rootProject.file('gradle/marathon_deployment.gradle')
 
 tasks.withType(Test) {
     environment << [
-            CONSUL_HOSTNAME                  : System.getenv("CONSUL_HOSTNAME") ?: testConsulHost,
-            ENVIRONMENT                      : "integration",
-            MOBILE_CONFIG_SERVICE_ENVIRONMENT: "testing",
-            EXAMPLE_SERVICE_ENVIRONMENT      : "testing",
-            DISCUSSION_SERVICE_ENVIRONMENT   : "testing",
+            CONSUL_HOSTNAME                        : System.getenv("CONSUL_HOSTNAME") ?:
```
Comment (sqreek) [inline gradle/dropwizard_service.gradle]: This part will be changed (here) [https://github.com/Wikia/pandora/pull/188]

File: service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/core/Problem.java (hunk 1 above)
Comment (sqreek) [inline service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/core/Problem.java]: :+1:

File: service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/exceptions/MobileApplicationsModuleNotSetException.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.wikia.services.mobileapplications.exceptions;
+
+public class MobileApplicationsModuleNotSetException extends Exception {
+
+  public static final String
+      EXPLANATION_MESSAGE =
```
Comment (sqreek) [inline service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/exceptions/MobileApplicationsModuleNotSetException.java]: line breaking is strange here

File: service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/guicemodule/MobileApplicationsModuleBase.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import io.dropwizard.client.HttpClientBuilder;
+import io.dropwizard.setup.Environment;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Named;
+
+public abstract class MobileApplicationsModuleBase extends AbstractModule {
```
Comment (sqreek) [inline service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/guicemodule/MobileApplicationsModuleBase.java]: Guice suggest to not use abstract in Module inherited from AbstractModule.
You can replace this using install(otherModule).

And now I figure it out I was left this here knowing about this need refactor

**Comment by sqreek** [comment]
except my own legacy code LGTM.
From my point of view you can merge it.
I can handle legacy code (Inherits from AbstractModule shouldn't be abstract)
@pchojnacki should take conflict part about Enviorement aproach.

File: service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/exceptions/ApplicationsNotFoundException.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import javax.ws.rs.core.Response;
+
+public class ApplicationsNotFoundException extends MobileApplicationsException {
+
+  static final String EXCEPTION_MESSAGE_FORMAT = "Applications for platform %s not found";
+
+  public ApplicationsNotFoundException(String platform) {
+    super(
+        Response.status(Response.Status.NOT_FOUND)
```
Comment (nmonterroso) [inline service/mobile-applications/src/main/java/com/wikia/services/mobileapplications/exceptions/ApplicationsNotFoundException.java]: I'd recommend using `StandardResponses` here

**Comment by nmonterroso** [comment]
minor comment otherwise LGTM

**Pull request merged**